### ISSUE TITLE
Cannot find Diffusion loss during training

### ISSUE THREAD

**Issue by championbridge** [JiuhaiChen/BLIP3o#7]
While rewriting the training code, I found that although the encoder in the blip3o framework initializes the DiT module, the loss computation in blip3o does not involve diffusion loss, which contradicts the diffusion loss described in the second stage of Sequential Training in the paper.

**Comment by JiuhaiChen** [comment]
Hi, Thanks for your interest! Please refer to https://github.com/JiuhaiChen/BLIP3o/blob/main/blip3o/model/language_model/blip3o_qwen.py#L164
this is the flow matching loss. Let me know if you have any questions.

**Comment by championbridge** [comment]
Thank you for your response. Just to clarify, is the training framework based on DiT, while the inference stage utilizes EmuGen's UNet?

**Comment by JiuhaiChen** [comment]
Yeah, in the inference time, the first diffusion process is based on our DiT, and we use EmuGen's UNet for the second stage diffusion (from CLIP feature to real image).

**Comment by championbridge** [comment]
Thank U！

**Comment by LighteningDarkness** [comment]
Does it mean that while training we only need to train the DiT with the flow-matching loss and not necessory to train the secong stage diffusion?

**Comment by LighteningDarkness** [comment]
> Yeah, exactly. We only tune the first stage DiT.
Thank you for answering my question! Really a NICE WORK!

**Comment by DOUDOU0314** [comment]
Hi~so when we train the second stage diffusion?
